src/test/java/org/hydev/mcpm/client/arguments/CommandsFactoryTest.java

```diff
@@ -0,0 +1,31 @@
+package org.hydev.mcpm.client.arguments;
+
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Tests for the CommandFactory
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-21
+ */
+class CommandsFactoryTest
+{
+    @Test
+    void test() throws ArgumentParserException
+    {
+        var ba = CommandsFactory.baseArgsParser();
+        var sa = CommandsFactory.serverArgsParser(System.out::println);
+
+        assert ba.getRawSubparsers().stream().anyMatch(it -> it.name().equals("list"));
+        assert ba.getRawSubparsers().stream().noneMatch(it -> it.name().equals("load"));
+        assert sa.getRawSubparsers().stream().anyMatch(it -> it.name().equals("list"));
+        assert sa.getRawSubparsers().stream().anyMatch(it -> it.name().equals("load"));
```

Comment:
These are regular `assert`s, you should use junit's Assertions class at the very least.

But that's not the key issue here. I think it's a little questionable whether or not CommandsFactoryTest can have proper tests. I don't think whether or not there is a sub parser with a certain name is an important thing to check (e.g. what if we change the name of the list command). I would consider whether or not ArgsParser can handle a `list` command to be important, but again that's more of an integration test.